plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -39,16 +39,35 @@ class CuraEngineBackend(Backend):
     #
     #   This registers all the signal listeners and prepares for communication
     #   with the back-end in general.
+    
+    executableName = "CuraEngine"
+    
     def __init__(self):
         super().__init__()
 
         # Find out where the engine is located, and how it is called.
         # This depends on how Cura is packaged and which OS we are running on.
-        default_engine_location = os.path.join(Application.getInstallPrefix(), "bin", "CuraEngine")
+        default_engine_location = None
+        if os.path.exists(os.path.join(Application.getInstallPrefix(), "bin", self.executableName)):
+            default_engine_location = os.path.join(Application.getInstallPrefix(), "bin", self.executableName)
         if hasattr(sys, "frozen"):
-            default_engine_location = os.path.join(os.path.dirname(os.path.abspath(sys.executable)), "CuraEngine")
+            default_engine_location = os.path.join(os.path.dirname(os.path.abspath(sys.executable)), self.executableName)
         if Platform.isWindows():
             default_engine_location += ".exe"
+        if Platform.isLinux() and not default_engine_location:
```

Comment:
Part of me thinks it is a shame to limit this to Linux. On the other hand, I do not think this happens often on the other two platforms.